Paralayout/ParalayoutAlerts.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  Copyright © 2025 Block, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import os
+
+private let ParalayoutLog = OSLog(subsystem: "com.squareup.Paralayout", category: "layout")
+
+/// Triggered when an alignment method is called that uses mismatched position types, i.e. aligning a view's leading or
+/// trailing edge to another view's left or right edge, or vice versa. This type of mismatch is likely to look correct
+/// under certain circumstance, but may look incorrect when using a different user interface layout direction.
+func ParalayoutAlertForMismatchedAlignmentPositionTypes() {
+    os_log(
```

Comment:
I'm a huge proponent of letting customers define their own error handling, [like so](https://github.com/dfed/Relativity/blob/016ab52094e75591d38ec9b1ea9a1489ccc387c2/Sources/Relativity/ErrorHandler.swift#L24-L49). (Note, that code was written before `#fileID` existed)

In our own adoption, I'd love to set this stuff to trigger a `Tracer.assertionFailure` that'll log in production and assert in debug. Or maybe just log in a way that bug reports can capture? I personally have found `os_log` to be utterly useless.

This was here before, though, so far from blocking.

Reply:
Additional thought: providing `#function` to consumers in the alert body of a closure allows the consumer to opt out of alerting on specific functions, which is kind of nice.

Reply:
In general agreed on that approach for the other alerts here (invalid hierarchy and mismatched positions).

This one is a little different in my mind though. We mostly hit this one because `bounds.size` hasn't been set yet (and so is still zero), but will self-resolve on the next layout pass after the size is set. So I don't think we actually want to `assertionFailure` on this one (or necessarily even add this to our logs, since it's noisy), but there's still value in being able to set the symbolic breakpoint if you see one that's _not_ self-resolving.

Reply:
> but there's still value in being able to set the symbolic breakpoint if you see one that's _not_ self-resolving

very aligned! good distinction here